src/main/java/echoserver/EchoClient.java

```diff
@@ -0,0 +1,16 @@
+package echoserver;
+
+import java.io.IOException;
+
+public class EchoClient implements Echoable {
```

Comment:
I am still apprehensive of using "client" here. The reason is that when we talk about a server, we inevitably end up talk about clients of the server as well. If we introduce the concept of a "client" in the server that is different from the clients of the server we end up with confusing language.

_Q: How does a client connect to your server?_
_A: Which client are you referring to?_

Reply:
The other reason is that it isn't really a client. It is kind of just a "mini-server"

Reply:
Okay I'll think of another term.  Perhaps thread?

Reply:
😄 I might think you are talking about OS threads. We will also be using threads do achieve concurrency.

Reply:
Just thought of another name - how about Echoer?